### PR TITLE
Add support for multiple coupons on Billing APIs

### PR DESCRIPTION
**Pull request by remi-stripe**
Add support for multiple coupons on Billing APIs
  * Add support for arrays of expandable API resources otherwise returning an array of strings by default
  * Add support for `Id`, `Invoice` and `InvoiceItem` on `Discount`.
  * Add support for `Discounts` on `Invoice`, `InvoiceItem`, `InvoiceLineItem`
  * Add support for `DiscountAmounts` on `CreditNote`, `CreditNoteLineItem`, `InvoiceLineItem`
  * Add support for `TotalDiscountAmounts` on `Invoice`

This PR supersedes https://github.com/stripe/stripe-dotnet/pull/2016 but used @ob-stripe's original commit as a base
It also mirrors https://github.com/stripe/stripe-java/pull/1070 in java

r? @richardm-stripe 
cc @stripe/api-libraries

### CODE REVIEW

File: src/StripeTests/Services/Accounts/AccountServiceTest.cs [hunk 1]
```diff
@@ -34,6 +34,17 @@ public class AccountServiceTest : BaseStripeTest
                     Name = "business name",
                 },
                 BusinessType = "company",
+                Capabilities = new AccountCapabilitiesOptions
```
Comment (richardm-stripe) [inline src/StripeTests/Services/Accounts/AccountServiceTest.cs:37]: Is this change necessary with the bump of stripe-mock version? (Just wondering why it's present in this PR)

Comment (remi-stripe) [reply src/StripeTests/Services/Accounts/AccountServiceTest.cs:37]: Yes! `RequestedCapabilities` is deprecated and was replaced with `Capabilities` last week so when we bumped stripe-mock` we had to make that change and change the `Type` on `AccountLink` too
